testsuite/features/step_definitions/api_common.rb

```diff
@@ -178,6 +178,36 @@
   assert_equal(false, ret.key?(attr))
 end
 
+Then(/^channel "([^"]*)" should (not )?be enabled on "([^"]*)"$/) do |channel, disabled, host|
+  node = get_target(host)
+  system_id = get_system_id(node)
+
+  channels = $api_test.channel.software.list_system_channels(system_id)
+  assert_equal(false, channels.empty?)
```

Comment:
are we sure it's an invalid case when this list is empty?

Reply:
uhm, I'd say this check may even be a problem if we want to check for a disabled channel on a minion which has no channels at all.

Probably it is worth to remove it from all the these step definitions, as it also redundant in the other cases.
`channels.includes` should return false anyway if the list is empty and checking a count against the channels.size will always work.